rstest_macros/src/parse/rstest/files.rs

```diff
@@ -15,21 +9,114 @@ use crate::{
 
 #[derive(Debug, Clone, PartialEq)]
 pub(crate) struct FilesGlobReferences {
+    base_dir: Option<LitStrAttr>,
     glob: Vec<LitStrAttr>,
     exclude: Vec<Exclude>,
     ignore_dot_files: bool,
+    ignore_missing_env_vars: bool,
 }
 
 impl FilesGlobReferences {
+    /// Replace environment variables in a string.
+    fn replace_env_vars(
+        &self,
+        attr: &LitStrAttr,
+        env: impl EnvironmentResolver,
+    ) -> Result<String, syn::Error> {
+        let re = Regex::new(r"\$([a-zA-Z_][a-zA-Z0-9_]+)|\$\{([^}]+)}")
+            .expect("Could not build the regex");
+
+        let ignore_missing = self.ignore_missing_env_vars;
+
+        let path = attr.value();
+        let haystack = path.as_str();
+        let mut result = String::with_capacity(attr.value().len());
+        let mut last_match = 0;
+
+        for caps in re.captures_iter(haystack) {
+            let match_all = caps.get(0).expect("The regex should have matched");
+
+            // Match the name of the variable and its default value (if any).
+            let (var_name, default) = if let Some(m) = caps.get(1) {
+                // In the first case ($VAR), the default value is None.
+                (m.as_str(), None)
+            } else {
+                // In the second case we have to split the variable name and the default value
+                // if there's a `:-` separator.
+                let m = caps.get(2).expect("The regex should have matched");
+
+                let (var_name, default) =
+                    if let Some((var_name, default)) = m.as_str().split_once(":-") {
+                        (var_name, Some(default))
+                    } else {
+                        (m.as_str(), None)
+                    };
+
+                var_name
+                    .chars()
+                    .all(|c| c.is_alphanumeric() || c == '_')
+                    .then_some((var_name, default))
+                    .ok_or_else(|| {
+                        attr.error(&format!(
+                            "The variable \"{}\" name does not match [a-zA-Z0-9_]+",
+                            m.as_str()
+                        ))
+                    })?
+            };
+
+            let v = env.get(var_name);
+            let replacement = match (v.as_ref().map(String::as_str), default) {
+                (Some(""), Some(default)) => default.to_string(),
+                (Some(val), _) => val.to_string(),
+                (None, Some(default)) => default.to_string(),
+                (None, _) if ignore_missing => String::new(),
+                (None, _) => {
+                    return Err(attr.error(&format!(
+                        "Could not find the environment variable {:?}",
+                        var_name
+                    )))
+                }
+            };
+
+            // Make sure cargo would rerun if the value of this environment variable changes.
+            println!("cargo::rerun-if-env-changed={var_name}");
```

Comment:
Unfortunately it doesn't seem to work. I tried it this feature for base root and if just change, add or remove this variable the tests are not recompiled. If I add if to a build script it seams to work.

I'll remove it and add some documentation notes.

Reply:
I tested it on my side, let me investigate a bit.